src/anyio/_core/_sockets.py

```diff
@@ -683,16 +683,30 @@ async def setup_unix_local_socket(
     path_str: str | bytes | None
     if path is not None:
         path_str = os.fspath(path)
+        is_abstract = (
+            path_str.startswith(b"\0")
+            if isinstance(path_str, bytes)
+            else path_str.startswith("\0")
+        )
 
-        # Copied from pathlib...
-        try:
-            stat_result = os.stat(path)
-        except OSError as e:
-            if e.errno not in (errno.ENOENT, errno.ENOTDIR, errno.EBADF, errno.ELOOP):
-                raise
+        if is_abstract:
```

Comment:
I really dislike no-op `if` branches. Please refactor the stat checking branch to be the only one. Move the comment about abstract namespaces to whatever place you deem most appropriate.

Reply:
Updated and removed extra var since the check is now more concise